metron-analytics/metron-profiler-common/src/main/java/org/apache/metron/profiler/DefaultMessageRouter.java

```diff
@@ -62,21 +68,45 @@ public DefaultMessageRouter(Context context) {
   @Override
   public List<MessageRoute> route(JSONObject message, ProfilerConfig config, Context context) {
     List<MessageRoute> routes = new ArrayList<>();
-    @SuppressWarnings("unchecked")
-    final Map<String, Object> state = (Map<String, Object>) message;
 
     // attempt to route the message to each of the profiles
     for (ProfileConfig profile: config.getProfiles()) {
+      Optional<MessageRoute> route = routeToProfile(message, profile);
+      route.ifPresent(routes::add);
+    }
 
+    return routes;
+  }
+
+  /**
+   * Creates a route if a message is needed by a profile.
+   * @param message The message that needs routed.
+   * @param profile The profile that may need the message.
+   * @return A MessageRoute if the message is needed by the profile.
+   */
+  private Optional<MessageRoute> routeToProfile(JSONObject message, ProfileConfig profile) {
+    Optional<MessageRoute> route = Optional.empty();
+
+    // allow the profile to access the fields defined within the message
+    @SuppressWarnings("unchecked")
+    final Map<String, Object> state = (Map<String, Object>) message;
+
+    try {
       // is this message needed by this profile?
       if (executor.execute(profile.getOnlyif(), state, Boolean.class)) {
 
         // what is the name of the entity in this message?
         String entity = executor.execute(profile.getForeach(), state, String.class);
-        routes.add(new MessageRoute(profile, entity));
+        route = Optional.of(new MessageRoute(profile, entity));
       }
+
+    } catch(Exception e) {
+      // log an error and move on. ignore bad profiles.
+      LOG.error(String.format("error encountered while executing profile; profile='%s', error='%s'", profile.getProfile(), e.getMessage()));
```

Comment:
I think `LOG.error` has a construction that takes a string and the exception.  Should this be: `LOG.error(String.format("error encountered while executing profile; profile='%s', error='%s'", profile.getProfile(), e.getMessage()), e);` so we can recover the stack trace in the log?